### PR TITLE
Add support for deletion keys in Map changelisteners

### PR DESCRIPTION
**Pull request by cmelchior**
- Add support for getting deleted keys in a the `MapChangeSet`
- Upgraded Gradle to 6.8.3 (last release before 7.*)
- Fixed problem with `installTransformer` step hanging on CI (I think) 
- With this patch, search indexes are kept when upgrading the fileformat. This affected a lot of unit tests.

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -236,7 +236,20 @@ def runBuild(buildFlags, instrumentationTestTarget) {
       if (isReleaseBranch) {
         signingFlags = "-PsignBuild=true -PsignSecretRingFile=\"${SIGN_KEY}\" -PsignPassword=${SIGN_KEY_PASSWORD}"
       }
-      sh "./gradlew assemble ${buildFlags} ${signingFlags} --stacktrace"
+      // Work around XXX by building each artifact independantly instead of using Gradle to
```
Comment (nhachicha) [inline Jenkinsfile:239]: forgot issue number?

File: realm/realm-library/src/androidTest/java/io/realm/RealmMigrationTests.java [hunk 2]
```diff
@@ -22,6 +22,7 @@
 import org.hamcrest.CoreMatchers;
 import org.junit.After;
 import org.junit.Before;
+import org.junit.Ignore;
```
Comment (nhachicha) [inline realm/realm-library/src/androidTest/java/io/realm/RealmMigrationTests.java:25]: remove

File: realm/realm-library/src/main/cpp/CMakeLists.txt [hunk 3]
```diff
@@ -204,7 +204,7 @@ endif()
 set(WARNING_CXX_FLAGS "-Werror -Wall -Wextra -pedantic -Wmissing-declarations \
     -Wempty-body -Wparentheses -Wunknown-pragmas -Wunreachable-code \
     -Wno-missing-field-initializers -Wno-unevaluated-expression -Wno-unreachable-code \
-    -Wno-c99-extensions")
+    -Wno-c99-extensions -Wno-dtor-name")
```
Comment (nhachicha) [inline realm/realm-library/src/main/cpp/CMakeLists.txt:207]: why adding this flag?

**Review by nhachicha — APPROVED** [review]
small comments

File: realm/realm-library/src/main/cpp/CMakeLists.txt (hunk 3 above)
Comment (cmelchior) [reply realm/realm-library/src/main/cpp/CMakeLists.txt:207]: Due to https://github.com/realm/realm-core/issues/4732#issuecomment-854473573

**Pull request merged**